### PR TITLE
Closes #1278: Refactoring ParagraphMappingFlexiblePage Class

### LINKED ISSUE #1278: ParagraphMappingFlexiblePage Migration Process Not Working Correctly
## Problem/Motivation
I am attempting to use the ParagraphMappingFlexiblePage migration plugin to map more than one nodes from various migration_lookups, as paragraphs, into a single page. What am I finding is that the last source specified will be migrated but all previous sources are ignored. The code doesn't really match how I expect the plugin to work.

### To Reproduce
Try a migration like this:
```
  temp_body_value:
    plugin: migration_lookup
    source: nid
    migration: ua_gc_paragraph

  temp_panel_value:
    plugin: migration_lookup
    source: nid
    migration: ua_gc_panel_group


  field_az_main_content:
    plugin: paragraphs_mapping_flexible_page
    source:
      - '@temp_body_value'
      - '@temp_panel_value'
```

## Proposed resolution
Opening a PR now that refactors the ParagraphMappingFlexiblePage class.

### Expected behavior

In the new implementation the array structure from the $value variable is iterated and mapped into multiple paragraphs.

### PR DESCRIPTION
**Pull request by rmarsh1ua**
## Description

Refactoring ParagraphMappingFlexiblePage class to correctly process the condition in which there is more than one source value. 

Actually I have no idea if this is intended functionality for this class so please discuss.

## Related Issue
https://github.com/az-digital/az_quickstart/issues/1278

## How Has This Been Tested?
Running this against my own migrations and finding that I'm able to successfully migrate data per the migration script in the linked issue.

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] I've added this Pull Request to the AZ Quickstart Github project and set it to "Review in progress".
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by rmarsh1ua** [comment]
I see there's some code standard issues that need to be fixed, it's easy enough. I'm more curious to know if the actual code changes are desirable.

**Comment by trackleft** [comment]
So when we migrate from one field with a multiple value to another field with multiple value, it works (migration_lookup), but if we migrate from multiple fields to a multiple value field it doesn’t.
My suspicion is that there is something going on with the delta. This PR might fix that issue.

**Comment by joeparsons** [comment]
Error from ProboCI console for "Migration Tests" step:
```
In MigrateToolsCommands.php line 983:
                                               
  az_node_flexible_page Migration - 4 failed.  
```

Also, I used ProboCI's ssh terrminal feature (under the "Advanced Options" dropdown) to run `drush migrate mmsg`:
```
root@179fa4b90404:/var/www/html/sites/migration# drush mmsg az_node_flexible_page
 -------------- ------------------- ------- ------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  Source ID(s)   Destination ID(s)   Level   Message                                                                                                                                                        
    
 -------------- ------------------- ------- ------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  47                                 1       Value is not a valid entity. (/src/az-quickstart-scaffolding/web/modules/contrib/entity_reference_revisions/src/Plugin/DataType/EntityReferenceRevisions.php:117)  
  51                                 1       Value is not a valid entity. (/src/az-quickstart-scaffolding/web/modules/contrib/entity_reference_revisions/src/Plugin/DataType/EntityReferenceRevisions.php:117)  
  53                                 1       Value is not a valid entity. (/src/az-quickstart-scaffolding/web/modules/contrib/entity_reference_revisions/src/Plugin/DataType/EntityReferenceRevisions.php:117)  
  54                                 1       Value is not a valid entity. (/src/az-quickstart-scaffolding/web/modules/contrib/entity_reference_revisions/src/Plugin/DataType/EntityReferenceRevisions.php:117)  
 -------------- ------------------- ------- -------------------------------------------------------------------------------------------------------------------------------------------------------------------
```

**Comment by joeparsons** [comment]
@rmarsh1ua We can get you a database export from the D7 kitten site that is used as the source for Probo migration tests if you want.

**Comment by rmarsh1ua** [comment]
@joeparsons Yea, I think I won't be able to make much progress without the database export. That would be great if you could send it!

**Comment by rmarsh1ua** [comment]
God, I really need to fix my Code sniffer settings ...

**Comment by rmarsh1ua** [comment]
This should be fixed now. The issue that was introduced came from how the previous iteration of the code was actually trying to process multi-dimensioned arrays in a fashion that was exactly specific to the migration that used this plugin. The new implementation should be able to handle arrays no matter how deeply the migration node ids are buried in the value array.

File: modules/custom/az_migration/src/Plugin/migrate/process/ParagraphMappingFlexiblePage.php [hunk 1]
```diff
@@ -19,24 +19,42 @@ class ParagraphMappingFlexiblePage extends ProcessPluginBase {
...
+   * @param array $main_content
+   *   An array of target_ids/target_revision_ids to return.
+   * @param array $value_array
+   *   The incoming array of values to transform and iterate.
+   */
+  private function populateMainContentArray(array &$main_content, array $value_array) {
+    foreach ($value_array as $item) {
+      if (
+      is_array($item) &&
+      count($item) == 2 &&
```
Comment (trackleft) [inline modules/custom/az_migration/src/Plugin/migrate/process/ParagraphMappingFlexiblePage.php:46]:
```suggestion
      count($item) === 2 &&
```

**Comment by joeparsons** [comment]
Checked that the only existing Quickstart migration that uses this plugin is this one (it's not used by the az_demo migrations after all):
https://github.com/az-digital/az_quickstart/blob/main/modules/custom/az_migration/migrations/migrate_plus.migration.az_node_flexible_page.yml

**Review by joeparsons — APPROVED** [review]
Verified that the console output of the Migration Test in ProboCI looks good for the affected migration:
```
[notice] Processed 11 items (11 created, 0 updated, 0 failed, 0 ignored) - done with 'az_node_flexible_page'
```

Also, checked the ProboCI migration site and verified that the imported Flexible page content is in tact:
https://8e1a8322-9ded-48ff-bd1f-ad3a4db2f07d--site-migration.probo.build/

I think we can address adding inline documentation to this plugin as part of #1305

**Review by trackleft — APPROVED** [review]

**Review by tadean — APPROVED** [review]

**Pull request merged**